test/e2e/specs/scroll-behavior.js

```diff
@@ -138,6 +139,15 @@ module.exports = {
         'scroll to anchor on load'
       )
 
+      // #2593 no crash on malformed URL
+      .url('http://localhost:8080//scroll-behavior')
```

Comment:
I saw your comment on the issue but entering through this URL is not correct because the `/scroll-behavior/` section is the base URL. Meaning that in a real scenario, the server should not serve this URL. It just happens to serve with the dev server because it's a simple fallback server that doesn't care about the base but in a real-world scenario, the fallback happens **after** `/scroll-behavior/`, the base URL. The server should actually 404 on this URL

Reply:
what if the base URL is `/`? our production setup is to serve our vue app on the root of the domain. I see what you're saying when the app root URL is something other than the root, but in that case shouldn't this still work?

Reply:
but then you are looking for `/scroll-behavior//` (or `//` if base is `/`)

Reply:
right - in this case the base is `//`. the `scroll-behavior` part of this URL doesn't really matter to the reproduction or the fix, IIRC there just needs to be something rendered in order to assert on it being there (since the crash generally causes nothing to be rendered at all).